src/java.base/share/classes/java/lang/Thread.java

```diff
@@ -1765,7 +1765,6 @@ public void interrupt() {
             }
         }
         interrupted = true;
-        interrupt0();  // inform VM of interrupt
```

Comment:
Is this required to interrupt itself which is always allowed, also is required when `nioBlocker` is `null`?

Reply:
yes, ive fixed that

Reply:
This removes the `interrupt0()` call for current thread which is also in OpenJ9 thread version.
The difference between OpenJDK and OpenJ9 threads is the `interrupt0()` for current thread is not synchronized w/ `interruptLock`, is it possible that is the cause of https://github.com/eclipse-openj9/openj9/issues/15491?
In addition, OpenJDK sets `interrupted = true` which is used by `isInterrupted()`.

Reply:
yes thats my assumption. Although I dont think this implementation is correct yet.